File: problemList/slidingWindow/solution/LC1004.java
package problemList.slidingWindow.solution;

public class LC1004 {
    public int longestOnes(int[] nums, int k) {
        int n = nums.length, ret = 0, cnt = 0, left = 0;
        for(int i = 0;i < n;i++){
            cnt += (1 - nums[i]);
            while(cnt > k){
                cnt -= (1 - nums[left]);
                left++;
            }
            ret = Math.max(ret, i - left + 1);
        }
        return ret;
    }
}

File: SolutionsWithCorCpp/problem_list_0x3f/sliding_window/solution/LC1004.cpp
#include <algorithm>
#include <iostream>
#include <unordered_map>
#include <vector>
using namespace std;

int longestOnes(vector<int> &nums, int k) {
    int n = nums.size(), r = 0, ret = 0, cnt = 0;
    for(int i = 0;r < n;r++){
        if(nums[r] == 0) cnt++;
        while(cnt > k && i < n){
            if(nums[i] == 0) cnt--;
            i++;
        }
        if(i <= r) ret = max(ret, r - i + 1);
    }
    return ret;
}